Terminology (#7)
solana-foundation/developer-content: docs/terminology.md

A front-end application that interacts with a Solana cluster.

> ## associated token account

Reviewer: Since  we are adding short hands as definitions, can you add one for `ata`?

> An account that stores tokens belonging to a particular `mint`. When sending

Reviewer: ```suggestion
An account that stores tokens belonging to a particular [mint](#mint). When sending
```

tokens, the sender can create associated token accounts for the recipient on
demand, so the recipient can recieve the token in the receipent's associated
token account.

## bank state

The result of interpreting all programs on the ledger at a given
[tick height](#tick-height). It includes at least the set of all
[accounts](#account) holding nonzero [native tokens](#native-token).

## base58

A way of representing binary values as strings. Unlike similar systems like
base64, base58 omits similar-appearing characters to avoid confusion. Solana

> commonly uses base58 to refer to [wallet addresses](#wallet-address)

Reviewer: ```suggestion
commonly uses base58 to refer to [account](#account) or [wallet addresses](#wallet-address)
```

## cross-program invocation (CPI)

> A call from one on-chain [program](@program) to another. For more information,

Reviewer: ```suggestion
A call from one on-chain [program](#program) to another. For more information,
```

The duration of time for which a [validator](#validator) is unable to
[vote](#ledger-vote) on another [fork](#fork).

## major unit

The 'main' unit of a currency. For example, among fiat currencies, the dollar
(for USD), pound (for GBP) and Euro (for EUR) are major units. Solana's SOL

> token has a major unit called the [Sol](#sol). Each Sol is worth 1 billion of

Reviewer: ```suggestion
token has a major unit called the [Sol](#sol). Each Sol is worth 1 billion
```

> [lamports](#lamport). Since computers aren't good at decimal maths, transactions

Reviewer: ```suggestion
[lamports](#lamport). Since computers aren't good at decimal math, transactions
```

[message formatting inside of transactions](./developing/programming-model/transactions.md#message-format)
here.

## memo

The [memo program](https://spl.solana.com/memo) provides a way to write text to
the blockchain. The Memo program is often added as an instruction to transfer
transactions as a text description by the sender of what the transfer is for.
For example, 'Thanks!❤️ '.

## minor unit

The lesser unit of a currency. Well known minor units include US cents (for
USD), pence (for GBP) and Eurocents (for EUR). Solana's Sol

> [cryptocurrency](#cryptocurrency) has a minor unit called the

Reviewer: cryptocurrency is not defined as a term in this doc, so it should not be linked
```suggestion
cryptocurrency has a minor unit called the
```

## private key

Another word for the [#secret-key]. Solana tools and documentation generally use
the term 'secret key' instead.

> ## program (or on-chain program)

Reviewer: ```suggestion
## program
```

Changing the heading will change the direct anchors to the content. Let's avoid doing so as much as possible

The component of a [validator](#validator) responsible for [program](#program)
execution.

## secret key

The secret key of a [keypair](#keypair).

## Sealevel

> Solana's parallel runtime for [on-chain programs](#on-chain-program).

Reviewer: ```suggestion
Solana's parallel runtime for [on-chain programs](#program).
```

`on-chain-program` is not a valid anchor on this page

## smart contract

> Another term for [on chain programs](#programs). Solana programs should use the

Reviewer: ```suggestion
Another term for [on chain programs](#program). Solana programs should use the
```

> term [on chain programs](#programs).

Reviewer: ```suggestion
term ["on-chain programs"](#program) or simply ["programs"](#program).
```

Author: Much better!

A collection of [keypairs](#keypair) that allows users to manage their funds.

## wallet address

The unique [public key](#public-key-pubkey) of a wallet. These are usually

> displayed to programs as a [base58](#base58) string.

Reviewer: ```suggestion
displayed to programs or user interfaced as a [base58](#base58) string.
```

> displayed to programs as a [base58](#base58) string.

Author: ```suggestion
displayed to programs or user interfaces as a [base58](#base58) string.
```

## wallet app

> An end-user program with a user interface to interfact with a blockchain. A

Reviewer: ```suggestion
An end-user program with a user interface (UI) to interact with a blockchain. A
```

wallet typically includes a UI to send and recieve tokens and NFTs, check

> balances of diffferent tokens, view NFTs, stake, and on.off ramp into fiat and

Reviewer: ```suggestion
balances of diffferent tokens, view NFTs, stake, and on/off ramp into fiat and
```
